### DISCUSSION TITLE
ID's no longer matching between CO and DocDB assets

### DISCUSSION THREAD

**Discussion by Sun-flow** [AllenNeuralDynamics/aind-data-access-api#51] [Ideas]
**Is your feature request related to a problem? Please describe.**
Historically, implementations of our CO and DocDB databases maintained alignment between the ID fields for each asset. (Ie, the `_id` field for a DocDB asset was the same value as the `id` field of the corresponding CO asset). This made lookup between the two databases very simple, and ensured that the right asset was being accessed at all times.

With the new database, this feature no longer exists. It seems that the only way to do something like this would be to parse the `location` field of the DocDB asset to get the CO id, which is not ideal. 

**Describe the solution you'd like**
I would like a return of the original functionality, or at least something comparable. My goal when requesting the improvements that have recently been made to our databasing was to have easy assurance that the data assets I was accessing were aligned, and that I was always editing/reading the correct asset. This seems to have gotten more difficult with the new system. I feel that ID alignment is an important and valuable feature of our database.

**Describe alternatives you've considered**
The information I am asking for is still derivable from components of the assets, ie I could parse the `location` string of the DocDB asset to get the CO ID. But, this adds room for error and forces me to maintain a mapping between the DocDB ID and the corresponding CO ID. It concerns me that this is not idiot-proof.

There exists an `external_links.Code Ocean` field which contains the desired information. This is probably acceptable, but adds extra steps. Data can be filtered from `CO id` -> `DocDBAsset.external_links.Code Ocean`, or the other way around. Currently not all assets contain this field, but they will soon.

**Comment by dyf** [comment]
I think we need to maintain separate identifiers here. Separating gives us the ability to detect when CO records are missing. We can add a canned method to query just for CO assets?
